src/components/TokenMigration/TokenMigrationStepper.tsx

```diff
@@ -54,48 +52,20 @@ const TokenMigrationStepper = ({ tokenMigrationConfig }: Props) => {
         dispatch({ type: tokenMigrationReducer.actionTypes.FORM_SUBMIT, payload: formData });
       }
     },
-    ...transactions.flatMap((tx: Required<TxParcel>, idx) => [
-      {
-        label: translateRaw('CONFIRM_TRANSACTION'),
-        backBtnText: tokenMigrationConfig.formTitle,
-        component: TokenMigrationMultiTx,
-        props: {
-          account,
-          isSubmitting,
-          transactions,
-          tokenMigrationConfig,
-          currentTxIdx: idx
-        },
-        actions: (_: ISimpleTxFormFull, goToNextStep: () => void) => {
-          if (transactions.length > 1) {
-            prepareTx(tx.txRaw);
-          } else {
-            goToNextStep();
-          }
-        }
-      },
-      {
-        label: '',
-        backBtnText: translateRaw('CONFIRM_TRANSACTION'),
-        component: account && WALLET_STEPS[account.wallet],
-        props: {
-          network: account && account.network,
-          senderAccount: account,
-          rawTransaction: tx.txRaw,
-          onSuccess: sendTx
-        }
-      }
-    ]),
-    {
-      label: tokenMigrationConfig.receiptTitle,
-      component: TokenMigrationReceipt,
-      props: {
-        amount: reducerState.amount,
-        tokenMigrationConfig,
-        account,
-        transactions
-      }
-    }
+    ...createSignConfirmAndBroadcastTxSteps({
+      transactions,
+      backStepTitle: tokenMigrationConfig.formTitle,
+      amount: reducerState.amount!,
+      account: reducerState.account!,
+      flowConfig: tokenMigrationConfig,
+      receiptTitle: tokenMigrationConfig.receiptTitle,
+      multiTxTitle: translateRaw('CONFIRM_TRANSACTION'),
+      isSubmitting,
+      receiptComponent: TokenMigrationReceipt,
+      multiTxComponent: TokenMigrationMultiTx,
```

Comment:
Can this never be a regular TX? Like membership is only 1?

Reply:
yeah it can be, but it hasn't been in master yet. so we can address it when it pops up.

Reply:
This is not the case and is a bug right now. There is only one tx when the approval is already done. I just discovered this when testing AAVE migration.